src/System.Windows.Forms/src/System/Windows/Forms/DataObject.DataStore.cs

```diff
@@ -235,18 +236,22 @@ public virtual string[] GetFormats(bool autoConvert)
                             Debug.Assert(cur != null, "GetMappedFormats returned null for '" + baseVar[i] + "'");
                             for (int j = 0; j < cur.Length; j++)
                             {
-                                formats.Add(cur[j]);
+                                if (!distinctFormats.Contains(cur[j]))
+                                {
+                                    distinctFormats.Add(cur[j]);
+                                }
```

Comment:
a `HashSet<string>` will have better performance and should behave the same, if you only add to it it will maintain the order in which you added items. Since this is a intermediate collection and at the end you are doing `ToArray` anyways I don't think it would hurt to switch the collection type here. Thoughts?

Reply:
I did not remember this. I have made the change and added a comment.

Reply:
Actually you can just add, it already does a check and returns whether something was added
```suggestion
                                distinctFormats.Add(cur[j]);
```

I'll also note that adding `null` to a `HashSet` will throw but looking at the code flow it doesn't seem possible to have null values here.

Reply:
Thank you @weltkante 

![image](https://user-images.githubusercontent.com/4403806/89480791-8e177b00-d7d9-11ea-8ced-7d374a743a4d.png)
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.hashset-1.add?view=netcore-3.1#returns